### ISSUE TITLE
Support for onClick to Create a timemarker

### ISSUE THREAD

**Issue by abramjstamper** [esnet/react-timeseries-charts#289]
Maintainers: I've evaluated many react graphing libraries in the past week and I think that this is both the best performing and well featured. Keep up the good work!

I'm not sure if this is something that is already implemented or not. I've browsed the docs and both open and closed issues, but I couldn't find what I was looking for.

Is it possible to add an onClick event to a line graph to create a static `TimeMarker`? I know there's lots of places where an onHover event is used to create a dynamic marker that follows the mouse ([Tracker](http://software.es.net/react-timeseries-charts/#/guide/annotations), [Currency](http://software.es.net/react-timeseries-charts/#/example/currency), & [Cycling](http://software.es.net/react-timeseries-charts/#/example/cycling)). However, I'm specifically looking for a way to click a spot on the graph, and have a static `TimeMarker` placed there until the user clears or the page is re-rendered. I know that the [TimeMarker](http://software.es.net/react-timeseries-charts/#/guide/annotations) component exists, but I'm not sure it's possible to dynamically create a `TimeMarker` through clicking the graph.

Any suggestions are appreciated on how to make this, or something similar work?

**Comment by abramjstamper** [comment]
I tried writing a TimeMarkers class so I could have multiple markers. In the TimeMarker documentation, there's no information listed about which props are required, and which aren't. Obviously a date is required, so I supplied a date, but still no dice. It doesn't seem like I can create it with the ChartContainer as a parent supplying the necessary information, and it appears to be an invalid child to the ChartContainer.

```
import React, { Component } from 'react';
import moment from 'moment';
import { TimeMarker } from 'react-timeseries-charts';

class TimeMarkers extends Component {

  renderMarker = (marker) => {
    return <TimeMarker time={moment(marker).toDate()}/>;
  }

  render() {
    return (
      <div>
        {this.props.markers.forEach((marker) => this.renderMarker(marker))}
      </div>
    );
  }
}

export default TimeMarkers;
```

Is the TimeMarker component only for internal use? It seems like the appropriate component for what I want to do.

**Comment by pjm17971** [comment]
If you're making your own chart, that generates an set of `TimeMarkers`, then you will need to pass the props that are injected into your `TimeMarkers` component into the `TimeMarker`. So basically everything here:
http://software.es.net/react-timeseries-charts/#/api/charts/TimeMarker

That's because the surrounding `ChartRow` and `ChartContainer` determine the size (width and height) and the scales (`timeScale` is the only one that matters here because the `yScale` isn't used for this marker). If you console.log the props in your render function you'll see what's passed in. Those basically need to be forwarded to the `TimeMarker` so that it knows all those things. If you are showing the Info Box then you'll have to pass `infoValues` in as well, or else disable it with `showInfoBox`. Otherwise, the props are optional because they have defaults (except as you pointed out the `time` itself)

**Comment by pjm17971** [comment]
BTW, for you first question, you should be able to just render a bunch of `TimeMarker`s within the `<Charts>` tags. So basically you'd use the tracker and its callback (`onTrackerChanged`) to get the changing position (time) of the tracker (the users cursor). Then you can also get a callback for a click `onBackgroundClick`. With those two events you can keep state on your end of where the user has clicked in the form of a list of times. When you re-render you should be able to add n TimeMarkers to the `<Charts>` tags, one for each time. I haven't tried something like that, but it should work.

**Comment by abramjstamper** [comment]
Aside from creating another child component of `ChartContainer` to get access to the internal props, do you have any way to access those internal props (eg: height, width)?

**Comment by abramjstamper** [comment]
My `TimeMarkers` component now looks like, and the parent is passing in the props like `{...this.props}`.
```
export default class TimeMarkers extends Component {

  renderMarker = (marker) => {
    return (
      <TimeMarker
        key={marker}
        {...this.props}
        time={moment(marker).toDate()}
      />
    );
  }
  render() {
    console.log(this.props);
    return (
      <div>
        {[...this.props.timeMarkers].map(marker => this.renderMarker(marker))}
      </div>
    );
  }
}
```

Parent:
```
<Charts>
  {charts}
  <TimeMarkers {...this.props} />                                          //doesn't work
  <TimeMarker {...this.props} time={moment(1531248416298).toDate()}/>      //works
</Charts>
```

I'm simply passing the props from the parent `<Charts>`, which does not appear to render any `<TimeMarker>`. I'm not sure if it's because it's in a div, but I don't ever get anything to show up on the page. I can verify that the render function is getting called with the correct value because I `console.log` props. Even if I move the `renderMarker` function into the same class as the `ChannelsChart`, I just get an error that props is undefined, even though I know for certain that is not true. 

I think what's complicating this more so is that I have a `ChannelsChart` similar to the cycling example. I would like to generate these `TimeMarkers` in `<ChartContainer>` so that they will be the appropriate height instead of being having a small cut in between each channel. However, when I move it down there I get an error about it not being an eligible child. I have since commented out the code checking on that, and I still don't have anything rendered, which is the same results as above. Thus, I'm not sure that I can create a `<TimeMarkers>` component, and have it display multiple components given the constraints of the current implementation of `<TimeMarker>`.

As some debugging, I went into the `node_modules/react-time-series/lib/components`. I changed line 122 & 123 to `console.log(posx)` and `console.log(this.props)`. I don't have any print out in the console, not even from a tracker (`this.state.tracker`) that I have working on the graph. It appears that `<TimeMarker>` isn't rendering inside of my `<TimeMarkers>` component. Even with my hard coded `<TimeMarker>` I don't get any console output. Any thoughts as to why this might be happening?

Any suggestions would be appreciated.
